Sources/Apodini/Components/Component.swift

```diff
@@ -25,8 +25,24 @@ public protocol Component {
     
     /// A function that is called when a request reaches the `Component`
     func handle() -> Self.Response
+    /// The default implementation simply wraps results from
+    /// `handle() -> Self.Response` into `.final(..)`.
+    func handle() -> Action<Self.Response>
 }
 
+extension Component {
+    /// The default implementation simply wraps results from
+    /// `handle() -> Self.Response` into `.final(..)`.
+    public func handle() -> Action<Self.Response> {
+        .final(handle())
+    }
+
+    /// Either this function or `handle() -> Action<Self.Response>`
+    /// have to be overwritten.
+    public func handle() -> Self.Response {
+        fatalError("Either handle() -> Encodable or handle() -> Action<Encodable> have to be implemented")
```

Comment:
This indirection of a handle method calling the other, and therefore having both implemented by an extension makes our DSL more prone for issues where the return type of handle is not Encodable and therefore the handle method is not inferred properly by swift

This is why I don't like the idea of having two handle methods, since it reverts one of our biggest wins from splitting Components and Handlers (before we even did the split 😂 )

Reply:
Should be solved with c1d8e5a